src/com/seafile/seadroid2/SSLSeafileSocketFactory.java

```diff
@@ -0,0 +1,185 @@
+package com.seafile.seadroid2;
+
+import android.os.Build;
+
+import java.io.IOException;
+import java.net.InetAddress;
+import java.net.Socket;
+import java.security.KeyManagementException;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import javax.net.ssl.KeyManager;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLSocket;
+import javax.net.ssl.SSLSocketFactory;
+import javax.net.ssl.TrustManager;
+
+/**
+ * Custom SSLSocketFactory.
+ *
+ * Used to manually select TLS protocol versions.
+ *
+ * based on:
+ * https://stackoverflow.com/questions/1037590/which-cipher-suites-to-enable-for-ssl-socket
+ */
+public class SSLSeafileSocketFactory extends SSLSocketFactory {
+
+    private SSLContext m_ctx;
+    private String[] m_ciphers;
+    private String[] m_protocols;
+
+    public SSLSeafileSocketFactory(KeyManager[] km, TrustManager[] tm, SecureRandom random) throws NoSuchAlgorithmException, KeyManagementException
+    {
+        m_ctx = SSLContext.getInstance("TLS");
+        m_ctx.init(km, tm, random);
+
+        m_protocols = GetProtocolList();
+        m_ciphers = GetCipherList();
+    }
+
+    public String[] getDefaultCipherSuites()
```

Comment:
Please follow the java code style of the project: the "{" should end in last line instead of begin a new line.